pkg/sql/logictest/testdata/logic_test/enums

```diff
@@ -56,11 +56,6 @@ SELECT 'Z'::public.int
 ----
 Z
 
-query T
-SELECT 'Z'::"int"
-----
-Z
-
```

Comment:
Why does this have to go away?

Reply:
I don't think its actually a valid test case! `pg_catalog` is in our search path, so we try `pg_catalog.int` first while performing name resolution, which is a valid type! It'll be slightly different than pg (in pg `int` is an alias for `int4`, so `pg_catalog.int` doesn't exist)  because I'm accepting all of our statically known types here, but we could extend this with a white list of what types are actually in pg_catalog or not to match up exactly.